### DISCUSSION TITLE
EDC vs DSC

### DISCUSSION THREAD

**Discussion by dilipdhankecha** [eclipse-edc/.github#7] [General]
How are [EDC](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector) and [IDS connector](https://github.com/International-Data-Spaces-Association/DataspaceConnector) similar? 

What are the differences between them and what would you recommend is better of the two? 

We are currently thinking of integrating one of these 2 in our product offering, and wanted to understand more on this. Also, if you can help us in understanding how CatenaX, GaiaX, TractusX etc. are linked to EDC and IDS, that would be helpful in terms of streamlining our approach. 

Thanks in advance!

**Comment by mspiekermann** [comment]
Hi @dilipdhankecha, 

***Tl;dr:*** The Eclipse Dataspace Connector (EDC) expands and improved the linked IDS-Connector (DSC) as a follow-up development by an OSS community, therefore from my side the clear recommendation to use the EDC.

***The longer story:*** the linked IDS connector (DSC) is a component previously developed by Fraunhofer ISST. We had the goal to build an IDS-compliant connector driven by a community (e.g., from initiatives you mentioned) as OSS to foster distribution and usage. While our discussions with several partner organizations begin/mid last year, it became very clear that the underlying governance and rules for contributions did not fit the requirements to form such a community and work on the code together due to questions regarding IP, licensing etc. 

As a solution, partners agreed in June '21 to create a project within the Eclipse Foundation (EF), since EF provides clear governance structures and rules allowing exactly the collaboration and community building, we are aiming for.
While the transfer process and meetings with initial committers (among others), we rethought the architecture to directly address future requirements in terms of applicability, robustness, scaling, etc. A much-discussed issue dealt with the limitation to a single data protocol such as the IDS vs. a solution that can support multiple such protocols. The decision fell on the latter and so the EDC is now developed based on a new architecture and design principles but based on available parts (like the IDS connector) that were transferred as initial contributions to the EF (see EF-project proposal). This time with a community of various companies (see https://projects.eclipse.org/projects/technology.dataspaceconnector/who) that bring in requirements from different initiatives like Catena-X, Eona-X, IDSA, Gaia-X, or MDS and work on a decentralized dataspace implementation based on a dataspace connector that provides capabilities for data exchange and ***extensibility to various shared service implementations*** (catalog, identity, usage control, logging, …) through interfaces. Thereby the EDC can support IDS(A)-based components like DAPS (Identity), or Metadata Broker (Catalog) as well as other approaches like WEB-DID (Identity) or Federated Catalog (Catalog). All with the same core to enable organizations to use a single technology to participate in multiple data spaces/ecosystems/federations in parallel.

Please have a look on the slides that were presented during the Gaia-X Hackathon event last week. I presented the above (esp. slide 13ff): https://github.com/eclipse-dataspaceconnector/Collateral/blob/main/Events/Hackathons/2022-03%20Gaia-X%20Hackathon%20No.%203/2022-03-28%20EDC-Basic.pdf 

How this is approach is implemented is well summarized in the session EDC Conceptual Overview and Architecture by Jim on the EDC Conference in January:  https://www.youtube.com/watch?v=IGd4oafLyAg&t=642s
 
> “Also, if you can help us in understanding how CatenaX, GaiaX, TractusX etc. are linked to EDC and IDS, that would be helpful in terms of streamlining our approach.”

- *Catena-X*: Uses the EDC as Connector component to build a dataspace between participants

- *Gaia-X*: EDC is an implementation to allow organizations to exchange data compliant to rules and policies of the Gaia-X AISBL according to the corresponding Trust Framework

- *Tractus-X*: Is an Eclipse Foundation OSS project initiated by Catena-X consortia, where specific (Catena-X related) EDC-extensions can be implemented under clear governance and rules by the EF.

The EDC thus ***expands and improved*** the capabilities of the DSC. It is being developed by an active OSS community project and is being contributed to the various initiatives that are involved in building data spaces.

The DSC is no longer maintained and further developed by Fraunhofer ISST. However, maintenance for current and potential users is offered by sovity GmbH in a corresponding fork (in the IDSA Github-Org).
